lib/dap/filter/simple.rb

```diff
@@ -171,6 +171,20 @@ def process(doc)
   end
 end
 
+class FilterFlatten
+  include Base
+  def process(doc)
+    self.opts.each_pair do |k,v|
+      if doc.has_key?(k) and doc[k].is_a?(Hash)
```

Comment:
I can see value in both ways -- flattening the entire structure or just selective parts or just to a certain depth.   

My initial thinking for this came from some of @tsellers-r7's recent data that is very nested and might benefit from this, and also that in Ruby, `flatten` on an array will flatten the entire thing unless given a depth argument and then it'll just flatten to that depth.